src/Integration/Binding/ProjectRuleSetWriter.cs

```diff
@@ -87,15 +87,21 @@ private bool TryUpdateExistingProjectRuleSet(string solutionRuleSetPath, string
         /// <summary>
         /// Remove all rule set inclusions which exist under the specified root directory.
         /// </summary>
-        internal /* testing purposes */  static void RemoveAllIncludesUnderRoot(RuleSet ruleSet, string rootDirectory)
+        internal /* testing purposes */  void RemoveAllIncludesUnderRoot(RuleSet ruleSet, string rootDirectory)
         {
             Debug.Assert(!string.IsNullOrWhiteSpace(rootDirectory));
-            
+
+            string ruleSetRoot = PathHelper.ForceDirectoryEnding(Path.GetDirectoryName(ruleSet.FilePath));
+
             // List<T> required as RuleSetIncludes will be modified
             List<RuleSetInclude> toRemove = ruleSet.RuleSetIncludes
                                             .Where(include =>
                                             {
-                                                string fullIncludePath = PathHelper.ResolveRelativePath(include.FilePath, rootDirectory);
+                                                string fullIncludePath = PathHelper.ResolveRelativePath(include.FilePath, ruleSetRoot);
```

Comment:
This `Where` clause is the part that was wrong. Should be resolving relative to the rule set file's root, NOT the 'removal' root (the directory under which any included rule set files should be removed).